Cargo.toml

```diff
@@ -12,16 +12,18 @@ license = "MIT"
 repository = "https://github.com/maoertel/diqwest"
 
 [features]
-default = []
+default = ["reqwest/default"]
 blocking = ["reqwest/blocking"]
+rustls-tls = ["reqwest/rustls-tls"]
 
 [dependencies]
 async-trait = "0.1"
 digest_auth = "0.3"
-reqwest = "0.11"
+reqwest = { version = "0.11", default-features = false }
 
-[target.'cfg(blocking)'.dependencies]
-reqwest =  { version = "0.11", features = ["blocking"] }
+# is this really needed?
+# [target.'cfg(blocking)'.dependencies]
+# reqwest =  { version = "0.11", features = ["blocking"] }
```

Comment:
Yes, this is needed for users that do not want to use the blocking part. So if you use it in default mode, the blocking feature of reqwest is not compiled.

Reply:
But "blocking" is not included in the default features of reqwest, only [default-tls](https://github.com/seanmonstar/reqwest/blob/5397d2cf8eaecc9f964ab4d2b40b8f2ad24ce26d/Cargo.toml#L29).

Reply:
Blocking is an optional feature of reqwest (see [here](https://docs.rs/reqwest/latest/reqwest/blocking/index.html#optional)), that is why it is optional in diqwest.

Reply:
Doesn't "reqwest/blocking" enable it already?

Reply:
Nope, that is the point. It is an optional feature, you have to enable it in your Cargo.toml, like:

`diqwest = { version = "...", features = ["blocking"] }`

Reply:
I get that. But I don't see why is that needed, because:

- diqwest depends on reqwest which by default does not have "blocking" enabled
- When you do diqwest = { ..., features=["blocking"] }, you're enabling "blocking" by saying "reqwest/blocking"

My point is, there's already the line:

[features]
blocking = ["reqwest/blocking"]

Then why is the commented one needed?

[Based on this](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features)

Reply:
Please read documentation. 

Again: blocking is a feature and is only enabled, when you configure it in your Cargo.toml - in both cases: reqwest & diqwest. When the feature is enabled in diqwest it uses [this](https://github.com/maoertel/diqwest/blob/9e2f1c9188aaed7eec6ea2e9529a22414a6fbc64/Cargo.toml#L25-L26) and compiles reqwest with the blocking feature. If the feature is enabled then it is used here in the code to enable the blocking part (see [here](https://github.com/maoertel/diqwest/blob/main/src/lib.rs#L42-L43))

Reply:
I've read the documentation. In fact: you are using [this](https://github.com/maoertel/diqwest/blob/9e2f1c9188aaed7eec6ea2e9529a22414a6fbc64/Cargo.toml#L25-L26):

`[target.'cfg(blocking)'.dependencies]`

When `target.cfg(arch).*` is for [platform specific instructions](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies). It specifically says:

> Unlike in your Rust source code, you cannot use [target.'cfg(feature = "fancy-feature")'.dependencies] to add dependencies based on optional features. Use [the [features] section](https://doc.rust-lang.org/cargo/reference/features.html) instead:

For example, see reqwest's [Cargo.toml](https://github.com/seanmonstar/reqwest/blob/e9ba0a9dc79f63c3655f334df23b50b9a841e326/Cargo.toml#L148)

What I'm saying is, what you want to achieve, you already achieved [here](https://github.com/maoertel/diqwest/blob/552daba0cf5c4f3a823145e1c2c5e13c710d8dc6/Cargo.toml#L16).

Reply:
Now I get your point, better I read documentation. Will look into it but probably after my vacation.

Reply:
Have a good one!